yt_dlp/extractor/youtube.py

```diff
@@ -793,7 +794,8 @@ def _extract_response(self, item_id, query, note='Downloading API JSON', headers
                     note='%s%s' % (note, ' (retry #%d)' % count if count else ''))
             except ExtractorError as e:
                 if isinstance(e.cause, network_exceptions):
-                    if isinstance(e.cause, compat_HTTPError):
+                    if isinstance(e.cause, compat_HTTPError) and not is_html(e.cause.read(512)):
+                        e.cause.seek(0)
```

Comment:
Sometimes the API errors may return a HTML page (e.g. dead endpoint).
`_parse_json` has no way of silently reporting the error, so this is another way I thought of.